include/nlohmann/detail/exceptions.hpp

```diff
@@ -131,7 +130,20 @@ class exception : public std::exception
         {
             return concat(a, '/', detail::escape(b));
         });
-        return concat('(', str, ") ");
+#if JSON_DIAGNOSTIC_POSITIONS
+        if ((leaf_element->start_pos() != std::string::npos) && (leaf_element->end_pos() != std::string::npos))
+        {
+            str += ", byte " + std::to_string(leaf_element->start_pos()) + "-" + std::to_string(leaf_element->end_pos());
+            return concat('(', str, ") ");
+        }
+#endif
```

Comment:
This logic looks weird:

```cpp
#if JSON_DIAGNOSTIC_POSITIONS
#endif
elif JSON_DIAGNOSTIC_POSITIONS
```

Reply:
Yes, a line was lost, and it should still be between the `#endif` and the `#elif`:

```
        return concat('(', str, ") ");
```